x/bsonx/bsoncore/array.go

```diff
@@ -81,18 +94,16 @@ func (a Array) String() string {
 
 	var elem Element
 	var ok bool
-	first := true
 	for length > 1 {
-		if !first {
-			buf.WriteByte(',')
-		}
 		elem, rem, ok = ReadElement(rem)
 		length -= int32(len(elem))
 		if !ok {
 			return ""
 		}
-		fmt.Fprintf(&buf, "%s", elem.String())
-		first = false
+		fmt.Fprintf(&buf, "%s", elem.Value().String())
+		if length != 1 {
```

Comment:
This makes sense. If the remaining length == 1, then we have read the last element (and the last byte is the trailing null byte).

Nit: I don't think we necessarily need to strictly validate here. But if the BSON array is malformed, I think remaining length could be < 1 here. E.g. if the trailing null byte was omitted, or if the array's length was inaccurate.

It might be reasonable to do:
```
if length > 1 {
  return ""
}
else if length == 1 {
  break
}
else {
  buf.WriteByte(',')
}
```

Reply:
Yeah, probably won't have perfect validation here. I assume you meant `length < 1`, but I agree/added these checks in. Added some test cases to ensure `""` is returned when array length is too long, too short, or the final null byte is missing.